src/Plugin.php

```diff
@@ -81,6 +81,10 @@ public static function init() {
     if (static::isPluginActive('bj-lazy-load/bj-lazy-load.php')) {
       add_filter('bjll/skip_classes', __CLASS__ . '::bjll_skip_classes');
     }
+
+    // Deletes transient on product plblish, update and delete.
+    add_action('before_delete_post', __NAMESPACE__ . '\Cache::flush_variation_attachments_transients', 10, 3);
+    add_action('woocommerce_update_product', __NAMESPACE__ . '\Cache::flush_variation_attachments_transients', 10, 3);
```

Comment:
What if the image is deleted or update? Shouldn't we also cover those cases?

Reply:
Yesterday I realised we are not pulling product images here, only variation images, which are not products by themselves. When you change a variation image, you still have to update the product, so this will trigger cache invalidation.

Reply:
I still have to check media deletion...